src/python/pants/engine/addressable.py

```diff
@@ -119,6 +119,9 @@ def __repr__(self):
       .format(type_constraint_type=type(self).__name__,
                     constrained_type=constrained_type))
 
+  def __name__(self):
+    return ', '.join(t.__name__ for t in self._types)
+
```

Comment:
this seems potentially bad - cant say I've ever heard of anyone overriding the `__name__` attribute (and I'm a little surprised you can - but hey, python). if you need to rely on this it'd probably be better to assign to a custom method/property vs overriding the interpreter-set, mostly magical `__name__`.